workers/unity/Packages/com.improbable.gdk.core/Utility/PlayerLoopUtils.cs

```diff
@@ -0,0 +1,238 @@
+                        if (!uniqueGroup.Contains(groupAttribute.GroupType))
+                        {
+                            uniqueGroup.Add(groupAttribute.GroupType);
+                            systems.Add(groupMgr);
+                        }
+                    }
+                }
+            }
+
+            // Sort all root groups, sorts depth first
+            foreach (var system in systems)
+            {
+                var type = system.GetType();
+                if (type == typeof(InitializationSystemGroup) ||
+                    type == typeof(SimulationSystemGroup) ||
+                    type == typeof(PresentationSystemGroup) ||
+                    type.GetCustomAttribute<UpdateInSubSystemAttribute>() != null)
+                {
+                    var groupSystem = system as ComponentSystemGroup;
+                    groupSystem.SortSystemUpdateList();
+                }
+            }
+        }
+
+        public static void AddToPlayerLoop(World world)
+        {
+            var systemGroups = world.Systems.OfType<ComponentSystemGroup>();
+            var subSystemToGroup = new Dictionary<Type, List<ComponentSystemGroup>>();
+
+            // Build lookup for PlayerLoop
+            foreach (var systemGroup in systemGroups)
+            {
+                var type = systemGroup.GetType();
+                Type subSystemType;
+
+                // Hardcode the subSystem types for Unity root-systems
+                if (type == typeof(InitializationSystemGroup))
+                {
+                    subSystemType = typeof(Initialization);
+                }
+                else if (type == typeof(SimulationSystemGroup))
+                {
+                    subSystemType = typeof(Update);
+                }
+                else if (type == typeof(PresentationSystemGroup))
+                {
+                    subSystemType = typeof(PreLateUpdate);
+                }
+                else
+                {
+                    var attributes = type.GetCustomAttributes(typeof(UpdateInSubSystemAttribute), true);
+                    if (attributes.Length == 0)
+                    {
+                        continue;
+                    }
+
+                    subSystemType = ((UpdateInSubSystemAttribute) attributes[0]).SubSystemType;
+                }
+
+                // Add to lookup table
+                if (!subSystemToGroup.TryGetValue(subSystemType, out var groupList))
+                {
+                    groupList = new List<ComponentSystemGroup>();
+                    subSystemToGroup.Add(subSystemType, groupList);
+                }
+
+                groupList.Add(systemGroup);
+            }
+
+            // Insert groups into PlayerLoop
+            var playerLoop = ScriptBehaviourUpdateOrder.CurrentPlayerLoop;
+            if (playerLoop.subSystemList == null)
+            {
+                playerLoop = PlayerLoop.GetDefaultPlayerLoop();
```

Comment:
`var playerLoop = ScriptBehaviourUpdateOrder.CurrentPlayerLoop ?? PlayerLoop.GetDefaultPlayerLoop();`

Reply:
I prefer to keep the long version here.